packages/zoe/src/cleanOfferRules.js

```diff
@@ -0,0 +1,52 @@
+import harden from '@agoric/harden';
+import { assert, details } from '@agoric/assert';
+
+export const cleanOfferRules = (keywords, amountMaths, offerRules) => {
```

Comment:
Sure, I think we can coerce each amount by the corresponding amountMath. What kind of validation can we do of the exit condition structure for `afterDeadline`? For `onDemand` and `waived` we can check the value is `null`.

Reply:
Adding Mark's comment here: 

> Well, at least that the timer and deadline properties exist. Yeah, when I wrote my review comment I hadn't realized how little we can say in general about timer and deadline being well formed. This makes sense though, because the well formedness of a given deadline is only wrt the timer that would understand that deadline.

Reply:
I've added a lot more validation. Let me know what you think